### ISSUE TITLE
Internal error for fully qualified `new` expressions with type specifiers

### ISSUE THREAD

**Issue by BryantLam** [chapel-lang/chapel#8555]
### Summary of Problem
Internal compiler error.
- Works if `record` is a `class`.
- Works if module prefix is omitted.

### Steps to Reproduce

**Source Code:**
```chpl
module A {
  record R {
    var a: int;

    proc init(a: int) {
      this.a = a + 1;
    }
  }
}

use A;
var r: A.R = new A.R(1); // Doesn't work.
/* var r: R = new R(1); // Works. */
writeln(r); // {a = 2}
```

### Configuration Information

`chpl version 1.17.0 pre-release (2f0af5c5db)`

**Comment by bradcray** [comment]
Thanks for filling this.  @lydia-duncan, @noakesmichael, @benharsh, does this failure mode look familiar to you?

I'm noting that the following variant works as well suggesting that it's specific to the `new A.R...` expression:

```chapel
var r: A.R = new R(1);
```

**Comment by cassella** [comment]
I found a few more variants that work, and one that results in a different error.

For conciseness, combining the ``use``, ``new``, and ``writeln``
statements onto one line, and then variably removing parts of it,

These fail as the original does (``internal error: assertion error [normalize.cpp:826]``):

```chapel
use A;  var r : A.R = new A.R(1);
use A;  var r : A.R = new A.R(1); writeln(r);
use A;  var r :   R = new A.R(1);
use A;  var r :   R = new A.R(1); writeln(r);
        var r : A.R = new A.R(1);
        var r : A.R = new A.R(1); writeln(r);
```

These work:

```chapel
use A;  var r       = new A.R(1);
use A;  var r       = new A.R(1); writeln(r);   // working variant with new A.R(1).
use A;  var r       = new   R(1);
use A;  var r       = new   R(1); writeln(r);
use A;  var r :   R = new   R(1); 
use A;  var r :   R = new   R(1); writeln(r);   // initial working example above
use A;  var r : A.R = new   R(1);
use A;  var r : A.R = new   R(1); writeln(r);   // Brad's working example above

        var r       = new A.R(1);
```


This fails to resolve ``R.writeThis``:

```chapel
        var r       = new A.R(1); writeln(r);
```

**Comment by lydia-duncan** [comment]
This is a variation on #6483, which notably did not include an explicit type for the declared variable and was resolved by #6810 and friends in July

**Comment by bradcray** [comment]
I renamed this issue to reflect Paul's observations and hopefully help make it easier to find in the future.  It seems that the key elements are using a fully qualified type name in a `new` expression (i.e., `new M.T(...)`) and using an explicit type specification for the resulting variable.

**Comment by BryantLam** [comment]
Due to changes to initializers in 1.17, the `super.init()` line in the original bug report is somewhat(?) irrelevant and can be removed for this specific use case.

**Comment by bradcray** [comment]
Removed the `super.init()` call from the original issue as Bryant suggests just above.

**Comment by bradcray** [comment]
@lydia-duncan , @benharsh: Any thoughts on what it would take to close this issue?  Whose court does it most logically belong in?

**Comment by lydia-duncan** [comment]
I would imagine this won't be too difficult to fix, just another instance of the initializer normalization not expecting a specific AST pattern.  I don't think it matters which one of us fixes it, but I would want to sync up with Ben before doing it myself

**Comment by benharsh** [comment]
I don't yet know what it would take to close this issue. I think it makes the most sense to assign it to me and let Lydia continue with python/extern-array stuff.

**Comment by bradcray** [comment]
@mppf: This seems like it might be related to your issue #10245.  Or to put another way, it seems as though if you were to fix #10245, you may find that you've fixed this as well.
